test/test_shape_ops.py

```diff
@@ -227,9 +227,8 @@ def test_diagonal_multidim(self, device, dtype):
         self.assertEqual(expected, result)
 
     @onlyNativeDeviceTypes
-    @dtypes(*get_all_dtypes(include_complex=False, include_bool=False, include_half=False,
-                            include_bfloat16=False))
-    @dtypesIfCUDA(*get_all_dtypes(include_complex=False, include_bool=False, include_bfloat16=False))
+    @dtypes(*all_types_and())
```

Comment:
We have just `all_types()`:

https://github.com/pytorch/pytorch/blob/aebde1bc2be708607cc9c6baf9fe402b368eb946/torch/testing/_legacy.py#L89